src/submission/ChildProcess.jl

```diff
@@ -72,9 +72,17 @@ end
 
 function launch_cmd(pt_arguments, exec_folder, dependencies, n_threads::Int, silence_mpi::Bool)
     julia_bin = Base.julia_cmd()
+    cur_proj  = Base.current_project()
+    if !isnothing(cur_proj)
+        # instantiate the project to make sure dependencies exist
+        # also, precompile to issues with coordinating access to compilecache
+        dir = dirname(cur_proj)
+        @info "forcing instantiate + precompile on project $dir"
+        run(`$julia_bin --project=$dir -e "using Pkg; Pkg.instantiate(); Pkg.precompile()"`)
+    end
     script_path = launch_script(pt_arguments, exec_folder, dependencies, silence_mpi)
-    return `$julia_bin 
-            --project   
+    return `$julia_bin
+            --project=$cur_proj
```

Comment:
Add support for out-of-project, or add a comment with a TODO

Reply:
done!